src/OrchardCore/OrchardCore.ResourceManagement/ResourceManager.cs

```diff
@@ -618,29 +672,15 @@ public ResourceTypeName(string resourceType, string resourceName)
 
             public bool Equals(ResourceTypeName other)
             {
-                if (other == null)
-                {
-                    return false;
-                }
-
-                return Type.Equals(other.Type) && Name.Equals(other.Name);
+                return Type == other.Type && Name == other.Name;
             }
 
             public override int GetHashCode()
             {
-                return Type.GetHashCode() << 17 + Name.GetHashCode();
+                return HashCode.Combine(Type,  Name);
             }
 
-            public override string ToString()
-            {
-                var sb = new StringBuilder();
-                sb.Append("(");
-                sb.Append(Type);
-                sb.Append(", ");
-                sb.Append(Name);
-                sb.Append(")");
-                return sb.ToString();
-            }
+            public override string ToString() => "(" + Type + ", " + Name + ")";
```

Comment:
StringBuilder can be an overkill too, concatenating < 10 strings usually is faster than doing the StringBuilder dance